library/ssl_tls12_server.c

```diff
@@ -2807,6 +2903,61 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
 #if defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED)
     if( ciphersuite_info->key_exchange == MBEDTLS_KEY_EXCHANGE_ECJPAKE )
     {
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+        unsigned char *out_p = ssl->out_msg + ssl->out_msglen;
+        unsigned char *end_p = ssl->out_msg + MBEDTLS_SSL_OUT_CONTENT_LEN -
+                               ssl->out_msglen;
+        psa_status_t status;
+        size_t output_offset = 0;
+        size_t output_len;
+        size_t ec_len;
+
+#if !defined(MBEDTLS_ECJPAKE_ALT)
```

Comment:
Note: we could probably solve both this and [that other comment](https://github.com/Mbed-TLS/mbedtls/pull/6533#discussion_r1022619707) by creating a global const buffer with the same content as `ecparameters` from `psa_pake_input_internal()` (except we should be using ` MBEDTLS_ECP_TLS_NAMED_CURVE` instead of hardcoding 3, I'm not sure why I didn't). Then here we only need to `memcpy()` from that buffer, and when reading we only need to `memcmp()` to it. And we avoid use of an `mbedtls_ecp_xxx()` function which we'd want to eliminate later.

The only drawback is duplication of a constant (`0x0023` for secp256r1) but that can probably be addressed later.